### PR TITLE
Handle WMS background layers dimensions + fixes

### PR DESCRIPTION
**Pull request by gberaudo**
Properly initialize background layers dimensions by passing the abstract
controller dimensions object.

Allow provisionning default dimensions with Angular gmfDefaultDimensions.
The dimensions should now be passed to the background selector.

Handle WMS layers in addition to WMTS layers.

### CODE REVIEW

File: contribs/gmf/src/services/themesservice.js [hunk 1]
```diff
@@ -224,6 +225,13 @@ gmf.Themes.prototype.getBgLayers = function() {
   };
 
   var layerLayerCreationFn = function(item) {
+    // Overwrite conflicting server dimensions with application ones
```
Comment (sbrunner) [inline contribs/gmf/src/services/themesservice.js]: We shouldn't declare the type if item and use the dot notation?

File: src/services/layerHelper.js [hunk 2]
```diff
@@ -49,18 +49,19 @@ ngeo.LayerHelper.REFRESH_PARAM = 'random';
 
 
 /**
- * Create and return a basic WMS layer with only a source URL and a dot
+ * Create and return a basic WMS layer with only a source URL and a cmoma
```
Comment (sbrunner) [inline src/services/layerHelper.js]: comma

File: contribs/gmf/src/services/themesservice.js (hunk 1 above)
Comment (gberaudo) [inline contribs/gmf/src/services/themesservice.js]: OK, I pushed some commits for adding types to externs and removing brackets.

File: src/services/layerHelper.js (hunk 2 above)
Comment (gberaudo) [inline src/services/layerHelper.js]: Fixed

**Comment by gberaudo** [comment]
@sbrunner, this branch should be ready now.
Do you want to test together with your changes for supporting WMS dimensions on c2cgeoportal before merging?

**Comment by gberaudo** [comment]
@sbrunner, OK to merge?

**Comment by sbrunner** [comment]
Looks good :-) merge

**Comment by gberaudo** [comment]
Thanks for the review.

**Pull request merged**